packages/react-cosmos-router-proxy/src/index.js

```diff
@@ -1,23 +1,51 @@
 import React from 'react';
-import { string } from 'prop-types';
+import { string, any } from 'prop-types';
 import { MemoryRouter, Route } from 'react-router';
 import { proxyPropTypes } from 'react-cosmos-shared/lib/react';
 import LocationInterceptor from './LocationInterceptor';
+import urlParser from 'url';
+
+function buildLocation(url, locationState) {
+  const parsedUrl = urlParser.parse(url);
```

Comment:
Nitpicking: Destructuring `const { pathname, search, hash } = urlParser.parse(url)` would make for clean shorthand keys in the returned object.

Reply:
Good idea, thanks.